### ISSUE TITLE
Bring back Tree for web accessibility reasons

### ISSUE THREAD

**Issue by AndreasSamjeske** [vaadin/flow#17988]
### Describe your motivation

Vaadin 8 had component **Tree**, which had pretty decent web accessibility features: aria-labelledby, aria-level and aria-expanded were automatically managed by Vaadin. Only drawback: every element was a <div>. <ul> and <li> would have been better.
### Describe the solution you'd like

Vaadin needs to have a tree, that consists of <ul> and <li>. This ensures users with screen readers are able to understand the structure easily.

### Describe alternatives you've considered

Since Vaadin 10+ there isn't any Tree component. In 2019 Vaadin [suggested to use TreeGrid with a single column](https://vaadin.com/labs/tree-component). But in regards of web accessibility this is insufficient: For screen reader users the grid structure is way too confusing and unnecessarily complex. 

In Vaadin directory there is a [tree](https://vaadin.com/directory/component/tree), too. But in HTML this is a grid as well.

### Additional context

[Accordion](https://vaadin.com/docs/latest/components/accordion) might be an alternative, if you have only one level of depth. But it lacks attribute aria-level and aria-expanded.
